src/main/java/hudson/remoting/Channel.java

```diff
@@ -1484,4 +1516,77 @@ public static Channel current() {
         // to avoid situations like this, create proxy classes that we need during the classloading
         jarLoaderProxy=RemoteInvocationHandler.getProxyClass(JarLoader.class);
     }
+
+    private static class ThreadLastIoId extends ThreadLocal<int[]> {
```

Comment:
IIRC this refactoring was to avoid a `$this` reference from an anonymous inner class, right? Might be worth it to mention that in Javadoc.